components/locale_core/src/preferences/mod.rs

```diff
@@ -529,6 +529,18 @@ macro_rules! __define_preferences {
             }
         }
 
+        impl From<$crate::preferences::LocalePreferences> for $name {
+            fn from(locale_prefs: $crate::preferences::LocalePreferences) -> Self {
```

Comment:
Issue: I don't know if I am onboard with this impl. It makes it easy to forget to forward the extensions from one preference bag to another preference bag. This is a source of bugs in ICU.

Reply:
that makes sense. I replaced it with a macro to generate conversions.